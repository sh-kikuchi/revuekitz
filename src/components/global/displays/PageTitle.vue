<script setup lang="ts">
import { computed } from 'vue'
const props = defineProps({
  id: {
    type: String,
    default: ''
  },
  class: {
    type: String,
    default: ''
  },
  style: {
    type: [String, Object],
    default: ''
  },
  styleReset: {
    type: Boolean,
    default: false
  }
})
//binding classes
const bindingClass = computed(() => {
  if (props.styleReset == true) {
    return props.class
  } else {
    return `revuekitz-title ${props.class} `
  }
})
</script>
<template>
  <h1 :id="props.id" :class="bindingClass" :style="props.style"><slot></slot></h1>
</template>
<style scoped>
.revuekitz-title {
  margin: 5px;
  text-align: center;
  font-size: 35px;
}
</style>
