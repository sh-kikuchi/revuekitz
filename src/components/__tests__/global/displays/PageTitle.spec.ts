import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import PageTitle from '../../../../components/global/displays/PageTitle.vue'

describe('PageTitle', () => {
  it('renders title correctly when styleReset is false', () => {
    const wrapper = mount(PageTitle, {
      props: {
        title: 'Test Title'
      }
    })

    expect(wrapper.find('h1').classes()).toContain('revuekitz-title')
  })

  it('renders title correctly when styleReset is true', () => {
    const wrapper = mount(PageTitle, {
      props: {
        title: 'Test Title',
        styleReset: true
      }
    })

    expect(wrapper.find('h1').classes()).not.toContain('revuekitz-title')
  })

  it('applies given id, class, and style props', () => {
    const wrapper = mount(PageTitle, {
      props: {
        id: 'test-id',
        class: 'custom-class',
        style: 'color: red;',
        title: 'Test Title'
      }
    })

    expect(wrapper.attributes('id')).toBe('test-id')
    expect(wrapper.classes()).toContain('custom-class')
    expect(wrapper.attributes('style')).toBe('color: red;')
  })
})
