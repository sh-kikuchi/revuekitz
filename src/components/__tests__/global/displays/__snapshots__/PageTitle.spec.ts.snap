// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle > matches snapshot 1`] = `"<h1 data-v-fa0f5afc="" id="" class="revuekitz-title" style=""></h1>"`;
